src/main/java/com/device/shop/controller/DiscountController.java

```diff
@@ -0,0 +1,51 @@
+package com.device.shop.controller;
+
+import com.device.shop.entity.Discount;
+import com.device.shop.exception.BadRequestException;
+import com.device.shop.service.DiscountService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@AllArgsConstructor
+public class DiscountController {
+
+    DiscountService discountService;
+
+    @PostMapping("/discount")
+    public ResponseEntity<Discount> addDiscount(@RequestBody Discount discount)  {
+        Discount addDiscount = discountService.addNewDiscount(discount);
+        return new ResponseEntity<>(addDiscount, HttpStatus.CREATED);
+    }
+
+    @GetMapping("/list/discount")
```

Comment:
Please change to `"/discount/list"`. It's better to have the same beginning in the urls in the same Controller